src/prefect/engine/task_runner.py

```diff
@@ -825,18 +826,27 @@ def get_task_run_state(self, state: State, inputs: Dict[str, Result]) -> State:
                     name=prefect.context.get("task_full_name", self.task.name)
                 )
             )
-            timeout_handler = prefect.utilities.executors.timeout_handler
+
+            # Note: Instead of using partial here, a contextlib.nullcontext could be
+            # used but this is not supported by py 3.6 so we will create this call
+            # then call it twice instead
+            run_task_with_timeout_handler = partial(
+                prefect.utilities.executors.run_task_with_timeout_handler,
+                task=self.task,
+                args=(),
+                kwargs=raw_inputs,
+                logger=self.logger,
+            )
```

Comment:
I actually asked @cicdw about doing that and he said that he would prefer that we establish a second use case before adding a new utility module since this is not about py3.6 compat for a feature but rather for pretty code.

Reply:
Fine. I'd argue that we should write code how we'd like to for the current version of python (when possible), and backport stdlib features we want to a `compatibility` module. A definition of `nullcontext` is like 3 lines, and IMO makes this code way more readable.

```python
from contextlib import contextmanager

@contextmanager
def nullcontext():
    pass

Reply:
Talked to Chris, could you add a small implementation of `nullcontext` in `prefect.utilities.compatibility` (importing from `contextlib` if present). No need for docs or tests on this, compat modules are short lived and non-public.

Reply:
Sounds good to me! Will do.

Reply:
For the record, my ultimate stance on this (independently of a new compat module) is that it's too much work / code movement all in service of removing a single duplicated line of code.  For reference, this was the duplicated line:
```python
value = timeout_handler(self.task.run, timeout=self.task.timeout, **raw_inputs)
```
I think DRY is useful when it tidies things up, but in this case it required a lot of triage, all for one line of code.  That being said, the new implementation isn't wrong so I'm not going to hold anything up for it.